pkg/workloads/manager.go

```diff
@@ -165,41 +165,109 @@ func (d *defaultManager) StopWorkloads(ctx context.Context, names []string) (*er
 		}
 	}
 
-	// Find all containers first
-	var containers []*rt.ContainerInfo
+	// Create an errgroup to handle both container and remote workloads
+	eg, ctx := errgroup.WithContext(ctx)
+
+	// Process each workload
 	for _, name := range names {
-		container, err := d.runtime.GetWorkloadInfo(ctx, name)
-		if err != nil {
-			if errors.Is(err, rt.ErrWorkloadNotFound) {
-				// Log but don't fail the entire operation for not found containers
-				logger.Warnf("Warning: Failed to stop workload %s: %v", name, err)
-				continue
-			}
-			return nil, fmt.Errorf("failed to find workload %s: %v", name, err)
-		}
+		name := name // capture loop variable
+		eg.Go(func() error {
+			return d.stopSingleWorkload(ctx, name)
+		})
+	}
 
-		running := container.IsRunning()
-		if !running {
-			// Log but don't fail the entire operation for not running containers
-			logger.Warnf("Warning: Failed to stop workload %s: %v", name, ErrWorkloadNotRunning)
-			continue
-		}
+	return eg, nil
+}
+
+// stopSingleWorkload stops a single workload (container or remote)
+func (d *defaultManager) stopSingleWorkload(ctx context.Context, name string) error {
+	// First, try to load the run configuration to check if it's a remote workload
+	runConfig, err := runner.LoadState(ctx, name)
+	if err != nil {
+		// If we can't load the state, it might be a container workload or the workload doesn't exist
+		// Try to stop it as a container workload
+		return d.stopContainerWorkload(ctx, name)
+	}
+
+	// Check if this is a remote workload
+	if runConfig.RemoteURL != "" {
+		return d.stopRemoteWorkload(ctx, name, runConfig)
+	}
+
+	// This is a container-based workload
+	return d.stopContainerWorkload(ctx, name)
+}
+
+// stopRemoteWorkload stops a remote workload
+func (d *defaultManager) stopRemoteWorkload(ctx context.Context, name string, runConfig *runner.RunConfig) error {
+	logger.Infof("Stopping remote workload %s...", name)
+
+	// Check if the workload is running by checking its status
+	workload, err := d.statuses.GetWorkload(ctx, name)
+	if err != nil {
+		// If we can't get the status, assume it's not running
+		logger.Warnf("Warning: Failed to get status for remote workload %s: %v", name, err)
+		return nil
+	}
+
+	if workload.Status != rt.WorkloadStatusRunning {
+		logger.Warnf("Warning: Remote workload %s is not running (status: %s)", name, workload.Status)
+		return nil
+	}
+
+	// Set status to stopping
+	if err := d.statuses.SetWorkloadStatus(ctx, name, rt.WorkloadStatusStopping, ""); err != nil {
+		logger.Warnf("Failed to set workload %s status to stopping: %v", name, err)
```

Comment:
I suggest adding more actionable info or setting this as a debug statement if it can't be actioned on.